database/buildSchema/19_application_status_history.sql

```diff
@@ -7,5 +7,24 @@ CREATE TABLE public.application_status_history (
     application_stage_history_id integer references public.application_stage_history(id),
     status public.application_status,
     time_created timestamp with time zone,
-    is_current bool
-);
\ No newline at end of file
+    is_current bool DEFAULT true
+);
+
+
+-- FUNCTION to set `is_current` to false on all other status_histories of current application
+CREATE OR REPLACE FUNCTION public.status_is_current_update()
+RETURNS trigger as $application_status_history_event$
+BEGIN
+	UPDATE public.application_status_history SET is_current = false
+	WHERE application_stage_history_id = NEW.application_stage_history_id AND id<>NEW.id;
```

Comment:
Okay, great -- I've used your auto-computed application_id column [below](https://github.com/openmsupply/application-manager-server/pull/94#discussion_r515854808) and modifed this function to use the new `application_id` column rather than `application_stage_history_id`.

So no gnarly query, and works regardless of connected stage. 👍